sdk/ml/azure-ai-ml/azure/ai/ml/_artifacts/_artifact_utilities.py

```diff
@@ -40,8 +36,6 @@
 from azure.ai.ml._utils._arm_id_utils import AMLNamedArmId
 from azure.ai.ml.constants import SHORT_URI_FORMAT, STORAGE_ACCOUNT_URLS
 from azure.ai.ml.entities._datastore._constants import WORKSPACE_BLOB_STORE
-from azure.ai.ml._artifacts._blob_storage_helper import BlobStorageClient
-from azure.ai.ml._artifacts._gen2_storage_helper import Gen2StorageClient
```

Comment:
I've removed any imports from this module that would require the presence of any of the Storage dependencies - this means that only the Storage type that's needed will be loaded - and only _when_ it needed.
This means that the non-default Storage providers (i.e. datalake and fileshare) can be removed from the dependencies.